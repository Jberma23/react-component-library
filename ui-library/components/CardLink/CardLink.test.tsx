import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { CardLink } from ".";

describe(CardLink, () => {
  it("renders with defaults", () => {
    const { asFragment } = render(
      <CardLink href="https://morningconsult.com/">test card</CardLink>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("appends additional classnames", () => {
    render(
      <CardLink className="my-test-class" href="https://morningconsult.com/">
        test card
      </CardLink>
    );
    const cardClass = screen.getByText("test card")?.getAttribute("class");
    expect(cardClass).toContain("my-test-class");
  });

  it("passes down additional props", async () => {
    const mockClickHandler = jest.fn();
    render(
      <CardLink onClick={mockClickHandler} href="https://morningconsult.com/">
        test card
      </CardLink>
    );
    const cardEl = screen.getByRole("link");
    await userEvent.click(cardEl);
    expect(mockClickHandler).toHaveBeenCalled();
  });

  it("can be disabled", () => {
    render(
      <CardLink href="https://morningconsult.com/" disabled>
        test card
      </CardLink>
    );
    const disabledAttr = screen
      .getByText("test card")
      ?.getAttribute("aria-disabled");
    expect(disabledAttr).toEqual("true");
  });
});
