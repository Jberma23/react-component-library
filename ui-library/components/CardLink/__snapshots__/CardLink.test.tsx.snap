// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardLink renders with defaults 1`] = `
<DocumentFragment>
  <a
    class="block rounded-3 hover:shadow-3 focus-visible:outline focus-visible:outline-2 focus-visible:outline-teal-500 cursor-pointer transition duration-default bg-white shadow-1"
    href="https://morningconsult.com/"
  >
    test card
  </a>
</DocumentFragment>
`;
